test_scheduling.cpp

```diff
@@ -283,6 +284,98 @@ int test_get_number_of_scheduled_actions()
     return num_errors;
 }
 
+int test_unit_enum_from_string()
+{
+    int num_errors = 0;
+    if (step_unit_enum_from_string("day") != StepUnit::Day)
+        num_errors++;
+    if (step_unit_enum_from_string("week") != StepUnit::Week)
+        num_errors++;
+    if (step_unit_enum_from_string("month") != StepUnit::Month)
+        num_errors++;
+    try {
+        step_unit_enum_from_string("invalid_input");
+        num_errors++;
+    }
+    catch (std::invalid_argument) {
+        // OK
+    }
+    catch (...) {
+        num_errors++;
+    }
+    return num_errors;
+}
+
+int test_output_schedule_from_string()
+{
+    int num_errors = 0;
+    Date st(2020, 1, 1);
+    Date end(2020, 1, 28);
+
+    Scheduler scheduling(st, end, StepUnit::Week, 1);
+    std::vector<bool> out = output_schedule_from_string(scheduling, "week", 0);
```

Comment:
Not having the zero here explicitly is probably better since the function already supports that. Nevertheless, one test of this case (unused but explicitly provided) should be included since that might be how the function will be used.